### PR TITLE
Bedrock fine tune

### PR DESCRIPTION
**Pull request by suyinw**
*Issue #, if available:*

*Description of changes:*
Create the contents for fine-tuning FMs in Bedrock with setup, fine-tune+evaluation, cleaning up the resources. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by markproy — COMMENTED** [review]
setup doesn't work for me:

× Building wheel for tokenizers (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [51 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-12-x86_64-cpython-311
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers
      copying py_src/tokenizers/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/models
      copying py_src/tokenizers/models/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/models
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/decoders
      copying py_src/tokenizers/decoders/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/decoders
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/normalizers
      copying py_src/tokenizers/normalizers/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/normalizers
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/pre_tokenizers
      copying py_src/tokenizers/pre_tokenizers/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/pre_tokenizers
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/processors
      copying py_src/tokenizers/processors/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/processors
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/trainers
      copying py_src/tokenizers/trainers/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/trainers
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/byte_level_bpe.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/sentencepiece_unigram.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/sentencepiece_bpe.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/base_tokenizer.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/char_level_bpe.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      copying py_src/tokenizers/implementations/bert_wordpiece.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/implementations
      creating build/lib.macosx-12-x86_64-cpython-311/tokenizers/tools
      copying py_src/tokenizers/tools/__init__.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/tools
      copying py_src/tokenizers/tools/visualizer.py -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/tools
      copying py_src/tokenizers/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers
      copying py_src/tokenizers/models/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/models
      copying py_src/tokenizers/decoders/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/decoders
      copying py_src/tokenizers/normalizers/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/normalizers
      copying py_src/tokenizers/pre_tokenizers/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/pre_tokenizers
      copying py_src/tokenizers/processors/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/processors
      copying py_src/tokenizers/trainers/__init__.pyi -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/trainers
      copying py_src/tokenizers/tools/visualizer-styles.css -> build/lib.macosx-12-x86_64-cpython-311/tokenizers/tools
      running build_ext
      running build_rust
      error: can't find Rust compiler
      
      If you are using an outdated pip version, it is possible a prebuilt wheel is available for this package but pip is not able to install from it. Installing from the wheel would avoid the need for a Rust compiler.
      
      To update pip, run:
      
          pip install --upgrade pip
      
      and then retry package installation.
      
      If you did intend to build this package from source, try installing a Rust compiler from your system package manager and ensure it is on the PATH during installation. Alternatively, rustup (available at https://rustup.rs/) is the recommended way to download and update the Rust compiler toolchain.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for tokenizers
ERROR: Could not build wheels for tokenizers, which is required to install pyproject.toml-based projects

**Review by markproy — COMMENTED** [review]
also setup uses `datasets` but doesn't install it

**Review by markproy — COMMENTED** [review]
I was using python 3.11.3 environment, and after running the pip installs, it somehow ended up w/ urllib in a bad state. thus s3 uploads of datasets would not work. to get uploads working again, I had to upgrade awscli package, which got urllib to urllib3-2.0.7 which seemed to bring s3 uploading back to life.

**Review by markproy — COMMENTED** [review]
nb 02 has this comment, but seems incorrect "# Select the foundation model you want to customize (you can find this from the "modelId" from listed foundation model above)". there is no list above

**Review by markproy — COMMENTED** [review]
in nb 02, remove the pip installs as they are dup of setup. if keeping those and the kernel restart, those should be the very first thing, not after the variable restores

**Comment by markproy** [comment]
> should add a markdown comment giving indication of how long the customization job should take

also same thing for Create PT which takes many minutes

**Comment by suyinw** [comment]
Thanks so much for the review @markproy and below are the edits to address the comments:
- Added dependency installation of tokenizers==0.12.1 to fix the installation issue of fmeval in 00_setup.ipynb; added setup tips with the full list of dependencies for reference
- Added installation of datasets in 00_setup.ipynb
- Removed the installation of fmeval package in 02 notebook as it's done in 00
- Removed the comment to select the modelId from the list, instead provided the reference model link from bedrock developer guide in 02 notebook
- The markdown for the estimated time of customization job and the PT job are already there (so left with no updates). 

Additionally edits:
- Added the code block in 02 nb to visualize the training/validation loss for fine-tuning.

**Review by ssinghgai — APPROVED** [review]
Please add instructions to notebook 02 to run Cleanup notebook OR add complete cleanup instructions in notebook 02.
